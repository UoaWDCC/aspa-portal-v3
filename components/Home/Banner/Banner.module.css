.title {
  font-size: 3.2vw;
  color: var(--aspa-white);
  font-weight: 400;
  line-height: 1.1;
  overflow: hidden;
}

.text {
  font-size: 1.4vw;
  line-height: 1.4;
  color: var(--aspa-white);
}

/* BUTTONS TO BE REPLACED LATER */
.button {
  width: 12vw;
  height: 4vw;
  font-size: 1.5vw;
  border-radius: 45px;
  background-color: #717882;
  font-weight: 600;
  color: #FBF5F0;
}

.button:hover {
  background-color: #8e949b;
}

.image {
  width: 60%;
  height: auto;
}

@media (max-width: 768px) {
  .title {
    font-size: 8vw;
  }

  .IntroductionStyles {
    text-align: center;
  }

  .text {
    font-size: 3vw;
  }

  .buttonGroup {
    flex-direction: column;
    align-items: center;
  }

  .button {
    width: 35vw;
    height: 8vw;
    font-size: 3vw;
  }
  
}