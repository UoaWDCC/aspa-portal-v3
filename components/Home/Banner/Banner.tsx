"use client";
import { Container, Title, Text, Stack, Box } from "@mantine/core";
import Image from "next/image";
import { motion } from "motion/react";
import styles from "./Banner.module.css";

export interface BannerProps {
  title: string;
  subtitle: string;
  description: string;
  paddingTop?: string;
  paddingBottom?: string;
  paddingleft?: string;
  paddingRight?: string;
  imageHeight?: string;
}

export function InfiniteScrollASPA() {
  const text = "ASPA ASPA ASPA ASPA ASPA ASPA ASPA ASPA ASPA ASPA";

  return (
    <div className={styles.scrollerWrapper}>
      <motion.div
        className={styles.scroller}
        animate={{ x: ["0%", "-50%"] }}
        transition={{
          repeat: Infinity,
          repeatType: "loop",
          duration: 20,
          ease: "linear",
        }}
      >
        <Title className={styles.movingAspa}>{text}</Title>
        <Title className={styles.movingAspa} style={{ marginLeft: "-5%" }}>
          {text}
        </Title>
      </motion.div>
    </div>
  );
}

export function Banner({
  title,
  subtitle,
  description,
  paddingTop = "",
  paddingBottom = "",
  paddingleft = "",
  paddingRight = "",
  imageHeight = "25vw",
}: BannerProps) {
  return (
    <Container
      fluid
      pt={paddingTop}
      pb={paddingBottom}
      pl={paddingleft}
      pr={paddingRight}
    >
      <Stack align="center" gap="sm" justify="center">
        <Box className={styles.imageContainer} style={{ height: imageHeight }}>
          <Image
            alt="ASPA_2025_Logo_Transparent"
            src="/ASPA_2025_Logo_Transparent.svg"
            width={500}
            height={500}
            className={styles.image}
          />
        </Box>
        <Title className={styles.title}>{title}</Title>
        <InfiniteScrollASPA />
        <Text className={styles.subtitle}>{subtitle}</Text>
        <Text className={styles.text}>{description}</Text>
      </Stack>
    </Container>
  );
}
