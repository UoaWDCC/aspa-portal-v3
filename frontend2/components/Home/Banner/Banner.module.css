@keyframes hoverUpDown {
    0%, 100% {
      transform: translateY(0);
    }
    50% {
      transform: translateY(-100px);
    }
  }
  
  .hoverUpDown {
    animation: hoverUpDown 2s ease-in-out infinite;
  }

  .title {
    font-size: 60px;
    margin-bottom: 15px;
    font-weight: 700;
    line-height: 1.1;
    overflow: hidden;
  }
  
  .group {
    color: #DD995B;
    margin-top: 15px;
  }
  
  .button {
    padding: 8px 16px;
    border-radius: 45px;
    background-color: #DD995B;
    font-weight: 600;
    margin: 15px 0;
    color: #FBF5F0;
  }
  
  .button:hover {
    background-color: #87562A;
  }
  
  .imageContainer {
    display: flex;
    align-items: center;
    justify-content: center;
  }
  
  .image {
    width: 100%;
    height: auto;
    margin-bottom: 1rem;
  }