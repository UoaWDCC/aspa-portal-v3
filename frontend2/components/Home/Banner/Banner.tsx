import { Container, Grid, Title, Text, Button, Group } from '@mantine/core';
import Image from "next/image";

interface BannerProps {
    title: string;
    description: string;
    buttonText: string;
    groupText: string;
    titleImageSrc: string;
    titleImageAlt: string;
    titleFontSize?: number;
    titleMarginBottom?: number;
    titleFontWeight?: number;
    titleLineHeight?: number;
    groupColor?: string;
    groupMarginTop?: number;
    buttonPadding?: string;
    buttonBorderRadius?: number;
    buttonBackgroundColor?: string;
    buttonFontWeight?: number;
    buttonMargin?: string;
    buttonColor?: string;
    buttonHoverBackgroundColor?: string;
    imageContainerDisplay?: string;
    imageContainerFlexDirection?: string;
    imageContainerAlignItems?: string;
    imageContainerJustifyContent?: string;
    imageWidth?: string;
    imageHeight?: string;
    imageMarginBottom?: string;
}

export function Banner({
    title,
    description,
    buttonText,
    groupText,
    titleImageSrc,
    titleImageAlt,
    titleFontSize = 60,
    titleMarginBottom = 15,
    titleFontWeight = 700,
    titleLineHeight = 1.1,
    groupColor = '#DD995B',
    groupMarginTop = 15,
    buttonPadding = '8px 16px',
    buttonBorderRadius = 45,
    buttonBackgroundColor = '#DD995B',
    buttonFontWeight = 600,
    buttonMargin = '15px 0',
    buttonColor = '#FBF5F0',
    buttonHoverBackgroundColor = '#87562A',
    imageContainerDisplay = 'flex',
    imageContainerAlignItems = 'center',
    imageContainerJustifyContent = 'center',
    imageWidth = '100%',
    imageHeight = 'auto',
    imageMarginBottom = '1rem'
}: BannerProps) {
  return (
    <Container>
      <Grid gutter="xl">
        <Grid.Col span={6}>
          <Title 
            order={1} 
            style={{
              fontSize: titleFontSize,
              marginBottom: titleMarginBottom,
              fontWeight: titleFontWeight,
              lineHeight: titleLineHeight,
              overflow: 'hidden'
            }}
          >
            {title}
          </Title>
          <Text>
            {description}
          </Text>
          <Group 
            align="center" 
            style={{
              color: groupColor,
              marginTop: groupMarginTop
            }}
          >
            <Button 
              style={{
                padding: buttonPadding,
                borderRadius: buttonBorderRadius,
                backgroundColor: buttonBackgroundColor,
                fontWeight: buttonFontWeight,
                margin: buttonMargin,
                color: buttonColor
              }}
              onMouseEnter={(e) => e.currentTarget.style.backgroundColor = buttonHoverBackgroundColor}
              onMouseLeave={(e) => e.currentTarget.style.backgroundColor = buttonBackgroundColor}
            >
                {buttonText}
            </Button>
            <Text>{groupText}</Text> 
          </Group>
        </Grid.Col>
        <Grid.Col 
          span={6} 
          style={{
            display: imageContainerDisplay,
            alignItems: imageContainerAlignItems,
            justifyContent: imageContainerJustifyContent
          }}
        >
          <Image 
            src={titleImageSrc} 
            alt={titleImageAlt} 
            width={0} 
            height={0} 
            style={{
              width: imageWidth,
              height: imageHeight,
              marginBottom: imageMarginBottom
            }} 
          />
        </Grid.Col>
      </Grid>
    </Container>
  );
}