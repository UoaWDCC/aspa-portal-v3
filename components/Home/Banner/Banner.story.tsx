import { Meta, StoryObj } from "@storybook/react";
import { Banner } from "./Banner";

const meta: Meta<typeof Banner> = {
  title: "Home/Banner",
  component: Banner,
  argTypes: {
    title: { control: "text" },
    description: { control: "text" },
    groupText: { control: "text" },
    titleImageSrc: { control: "text" },
    titleImageAlt: { control: "text" },
  },
};

export default meta;

export const Default: StoryObj<typeof Banner> = {
  args: {
    title: "Welcome to Auckland Student Pool Association!",
    description:
      "Here you can register to become a member of the club, log in to your account, and sign up for club events. Check out the list of current events to see what we’ve planned for the future, or have a look at our past events to see what we’re all about and some of the highlights of past years.",
    titleImageSrc: "/aspa_title.svg",
    titleImageAlt: "ASPA Title",
  },
};
