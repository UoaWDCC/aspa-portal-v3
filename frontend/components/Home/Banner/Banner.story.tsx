import { Meta, StoryObj } from '@storybook/react';
import { Banner } from './Banner';

const meta: Meta<typeof Banner> = {
    title: 'Home/Banner',
    component: Banner,
    argTypes: {
        title: { control: 'text' },
        description: { control: 'text' },
        buttonText: { control: 'text' },
        groupText: { control: 'text' },
        titleImageSrc: { control: 'text' },
        titleImageAlt: { control: 'text' }
    }
};

export default meta;


export const Default: StoryObj<typeof Banner> = {
    args: {
      title: 'Welcome to Auckland Student Pool Association!',
      description: 'Leque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, Leque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, Leque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur',
      buttonText: 'Current Events',
      groupText: 'dolor sit amet, consectetur',
      titleImageSrc: '/aspa_title.svg',
      titleImageAlt: 'ASPA Title'
    }
  };
