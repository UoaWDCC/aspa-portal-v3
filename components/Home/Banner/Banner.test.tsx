import { render, screen } from "@/test-utils";
import { Banner, BannerProps } from "./Banner";

describe('Banner Component Testing', () => {
  const originalInnerWidth = global.innerWidth;
  const longBannerProps : BannerProps = {
    title: "ASPA - Auckland Student Pool Association of the University of Auckland, New Zealand, Waipapa Taumata Rau",
    description: "Leque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, Leque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, Leque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur",
    buttonText: "View Auckland Student Pool Association of the the University of Auckland Page",
    groupText: "Button to view the Auckland Student Pool Association of the the University of Auckland Page",
    titleImageSrc: "/aspa_title.svg",
    titleImageAlt: "ASPA Title"
  }
  const bannerProps : BannerProps = {
    title: "ASPA - Auckland Student Pool Association",
    description: "Leque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, Leque porro quisquam est qui dolorem ipsum quia dolor sit amet",
    buttonText: "View ASPA Page",
    groupText: "Click Here To View ASPA Page",
    titleImageSrc: "/aspa_title.svg",
    titleImageAlt: "ASPA Title"
  }

  beforeEach(() => {
    global.innerWidth = originalInnerWidth;
    global.dispatchEvent(new Event('resize'));
  })

  it("should display all fields properly despite length", () => {
    
    render(
        <Banner 
          {...longBannerProps}
        />
    );
    expect(screen.getByText(longBannerProps.title)).toBeInTheDocument();
    expect(screen.getByText(longBannerProps.description)).toBeInTheDocument();
    expect(screen.getByText(longBannerProps.buttonText)).toBeInTheDocument();
    expect(screen.getByText(longBannerProps.groupText)).toBeInTheDocument();
    const img = screen.getByAltText(longBannerProps.titleImageAlt);
    expect(img).toBeInTheDocument();
    expect(img).toHaveAttribute("src", longBannerProps.titleImageSrc);
  })

  it("should render the logo image with the moving animation class", () => {
    render(
      <Banner
        {...bannerProps}
      />
    );
    const img = screen.getByAltText(bannerProps.titleImageAlt);
    expect(img).toBeInTheDocument();
    expect(img).toHaveClass("hoverUpDown");
  });

  it("should resize to fit mobile views after starting in desktop view" , () => {
    
    const { rerender } = render(
        <Banner 
          {...bannerProps}
        />
    );

    expect(screen.getByText(bannerProps.title)).toBeInTheDocument();

    global.innerWidth = 375;
    global.dispatchEvent(new Event('resize'));
    rerender(
      <Banner 
        {...bannerProps}
      />
    );

    expect(screen.getByText(bannerProps.title)).toBeInTheDocument();
    expect(screen.getByText(bannerProps.description)).toBeInTheDocument();
    expect(screen.getByText(bannerProps.buttonText)).toBeInTheDocument();
    expect(screen.getByText(bannerProps.groupText)).toBeInTheDocument();
    const img = screen.getByAltText(bannerProps.titleImageAlt);
    expect(img).toBeInTheDocument();
    expect(img).toHaveAttribute("src", bannerProps.titleImageSrc);
  });
  
  it("should match snapshot", () => {
    const title = "ASPA - Auckland Student Pool Association";
    const description = "Leque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, Leque porro quisquam est qui dolorem ipsum quia dolor sit amet";
    const buttonText = "View ASPA Page";
    const groupText = "View ASPA Page";
    const imageAlt = "ASPA Title";
    const { asFragment } = render(
      <Banner 
          title={title}
          description={description}
          buttonText={buttonText}
          groupText={groupText}
          titleImageSrc="/aspa_title.svg"
          titleImageAlt={imageAlt}
        />
    );
    expect(asFragment()).toMatchSnapshot();
  });
})
