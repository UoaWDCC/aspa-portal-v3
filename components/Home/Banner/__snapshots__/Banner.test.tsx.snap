// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner Component Testing Snapshot test for default banner 1`] = `
<DocumentFragment>
  <style
    data-mantine-styles="true"
  >
    :root{--mantine-font-family: fontFamily;--mantine-font-family-headings: fontFamily;--mantine-color-aspa-orange-0: #fff2e4;--mantine-color-aspa-orange-1: #fbe6d3;--mantine-color-aspa-orange-2: #f0ccaa;--mantine-color-aspa-orange-3: #e5af7e;--mantine-color-aspa-orange-4: #dc9758;--mantine-color-aspa-orange-5: #d78840;--mantine-color-aspa-orange-6: #d68032;--mantine-color-aspa-orange-7: #bd6d24;--mantine-color-aspa-orange-8: #a9611d;--mantine-color-aspa-orange-9: #945212;--mantine-color-aspa-brown-0: #faf5ef;--mantine-color-aspa-brown-1: #f0e8e0;--mantine-color-aspa-brown-2: #e2cdbb;--mantine-color-aspa-brown-3: #d5b192;--mantine-color-aspa-brown-4: #c9996f;--mantine-color-aspa-brown-5: #c38a59;--mantine-color-aspa-brown-6: #c1834d;--mantine-color-aspa-brown-7: #a96f3d;--mantine-color-aspa-brown-8: #986335;--mantine-color-aspa-brown-9: #845429;--mantine-color-aspa-gray-0: #f4f5f5;--mantine-color-aspa-gray-1: #e7e7e7;--mantine-color-aspa-gray-2: #cdcdcd;--mantine-color-aspa-gray-3: #b1b1b1;--mantine-color-aspa-gray-4: #999999;--mantine-color-aspa-gray-5: #8a8a8a;--mantine-color-aspa-gray-6: #838383;--mantine-color-aspa-gray-7: #707071;--mantine-color-aspa-gray-8: #636466;--mantine-color-aspa-gray-9: #52575b;}:root[data-mantine-color-scheme="dark"]{--mantine-color-aspa-orange-text: var(--mantine-color-aspa-orange-4);--mantine-color-aspa-orange-filled: var(--mantine-color-aspa-orange-8);--mantine-color-aspa-orange-filled-hover: var(--mantine-color-aspa-orange-9);--mantine-color-aspa-orange-light: rgba(214, 128, 50, 0.15);--mantine-color-aspa-orange-light-hover: rgba(214, 128, 50, 0.2);--mantine-color-aspa-orange-light-color: var(--mantine-color-aspa-orange-3);--mantine-color-aspa-orange-outline: var(--mantine-color-aspa-orange-4);--mantine-color-aspa-orange-outline-hover: rgba(220, 151, 88, 0.05);--mantine-color-aspa-brown-text: var(--mantine-color-aspa-brown-4);--mantine-color-aspa-brown-filled: var(--mantine-color-aspa-brown-8);--mantine-color-aspa-brown-filled-hover: var(--mantine-color-aspa-brown-9);--mantine-color-aspa-brown-light: rgba(193, 131, 77, 0.15);--mantine-color-aspa-brown-light-hover: rgba(193, 131, 77, 0.2);--mantine-color-aspa-brown-light-color: var(--mantine-color-aspa-brown-3);--mantine-color-aspa-brown-outline: var(--mantine-color-aspa-brown-4);--mantine-color-aspa-brown-outline-hover: rgba(201, 153, 111, 0.05);--mantine-color-aspa-gray-text: var(--mantine-color-aspa-gray-4);--mantine-color-aspa-gray-filled: var(--mantine-color-aspa-gray-8);--mantine-color-aspa-gray-filled-hover: var(--mantine-color-aspa-gray-9);--mantine-color-aspa-gray-light: rgba(131, 131, 131, 0.15);--mantine-color-aspa-gray-light-hover: rgba(131, 131, 131, 0.2);--mantine-color-aspa-gray-light-color: var(--mantine-color-aspa-gray-3);--mantine-color-aspa-gray-outline: var(--mantine-color-aspa-gray-4);--mantine-color-aspa-gray-outline-hover: rgba(153, 153, 153, 0.05);}:root[data-mantine-color-scheme="light"]{--mantine-color-aspa-orange-text: var(--mantine-color-aspa-orange-filled);--mantine-color-aspa-orange-filled: var(--mantine-color-aspa-orange-6);--mantine-color-aspa-orange-filled-hover: var(--mantine-color-aspa-orange-7);--mantine-color-aspa-orange-light: rgba(214, 128, 50, 0.1);--mantine-color-aspa-orange-light-hover: rgba(214, 128, 50, 0.12);--mantine-color-aspa-orange-light-color: var(--mantine-color-aspa-orange-6);--mantine-color-aspa-orange-outline: var(--mantine-color-aspa-orange-6);--mantine-color-aspa-orange-outline-hover: rgba(214, 128, 50, 0.05);--mantine-color-aspa-brown-text: var(--mantine-color-aspa-brown-filled);--mantine-color-aspa-brown-filled: var(--mantine-color-aspa-brown-6);--mantine-color-aspa-brown-filled-hover: var(--mantine-color-aspa-brown-7);--mantine-color-aspa-brown-light: rgba(193, 131, 77, 0.1);--mantine-color-aspa-brown-light-hover: rgba(193, 131, 77, 0.12);--mantine-color-aspa-brown-light-color: var(--mantine-color-aspa-brown-6);--mantine-color-aspa-brown-outline: var(--mantine-color-aspa-brown-6);--mantine-color-aspa-brown-outline-hover: rgba(193, 131, 77, 0.05);--mantine-color-aspa-gray-text: var(--mantine-color-aspa-gray-filled);--mantine-color-aspa-gray-filled: var(--mantine-color-aspa-gray-6);--mantine-color-aspa-gray-filled-hover: var(--mantine-color-aspa-gray-7);--mantine-color-aspa-gray-light: rgba(131, 131, 131, 0.1);--mantine-color-aspa-gray-light-hover: rgba(131, 131, 131, 0.12);--mantine-color-aspa-gray-light-color: var(--mantine-color-aspa-gray-6);--mantine-color-aspa-gray-outline: var(--mantine-color-aspa-gray-6);--mantine-color-aspa-gray-outline-hover: rgba(131, 131, 131, 0.05);}
  </style>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="m_7485cace mantine-Container-root"
    data-fluid="true"
  >
    <style
      data-mantine-styles="inline"
    >
      .__m__-«rr»{--grid-gutter:var(--mantine-spacing-xl);}
    </style>
    <div
      class="m_410352e9 mantine-Grid-root __m__-«rr»"
    >
      <div
        class="m_dee7bd2f mantine-Grid-inner"
      >
        <style
          data-mantine-styles="inline"
        >
          .__m__-«rt»{--col-flex-grow:auto;--col-flex-basis:100%;--col-max-width:100%;}@media(min-width: 48em){.__m__-«rt»{--col-flex-grow:auto;--col-flex-basis:50%;--col-max-width:50%;}}
        </style>
        <div
          class="m_96bdd299 mantine-Grid-col __m__-«rt»"
        >
          <h1
            class="title m_8a5d1357 mantine-Title-root"
            data-order="1"
            style="--title-fw: var(--mantine-h1-font-weight); --title-lh: var(--mantine-h1-line-height); --title-fz: var(--mantine-h1-font-size);"
          >
            ASPA - Auckland Student Pool Association
          </h1>
          <p
            class="mantine-focus-auto text m_b6d8b162 mantine-Text-root"
          >
            Leque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, Leque porro quisquam est qui dolorem ipsum quia dolor sit amet
          </p>
          <div
            class="m_4081bf90 mantine-Group-root"
            style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: flex-start; --group-wrap: wrap;"
          >
            <button
              class="mantine-focus-auto mantine-active button m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
              style="--button-color: var(--mantine-color-white);"
              type="button"
            >
              <span
                class="m_80f1301b mantine-Button-inner"
              >
                <span
                  class="m_811560b9 mantine-Button-label"
                >
                  View ASPA Page
                </span>
              </span>
            </button>
            <p
              class="mantine-focus-auto group m_b6d8b162 mantine-Text-root"
            >
              View ASPA Page
            </p>
          </div>
        </div>
        <style
          data-mantine-styles="inline"
        >
          .__m__-«r15»{--col-flex-grow:auto;--col-flex-basis:100%;--col-max-width:100%;}@media(min-width: 48em){.__m__-«r15»{--col-flex-grow:auto;--col-flex-basis:50%;--col-max-width:50%;}}
        </style>
        <div
          class="m_96bdd299 mantine-Grid-col imageContainer __m__-«r15»"
        >
          <img
            alt="ASPA Title"
            class="image hoverUpDown"
            data-nimg="1"
            decoding="async"
            height="0"
            loading="lazy"
            src="/aspa_title.svg"
            style="color: transparent;"
            width="0"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Banner Component Testing should match snapshot 1`] = `
<DocumentFragment>
  <style
    data-mantine-styles="true"
  >
    :root{--mantine-font-family: fontFamily;--mantine-font-family-headings: fontFamily;--mantine-color-aspa-orange-0: #fff2e4;--mantine-color-aspa-orange-1: #fbe6d3;--mantine-color-aspa-orange-2: #f0ccaa;--mantine-color-aspa-orange-3: #e5af7e;--mantine-color-aspa-orange-4: #dc9758;--mantine-color-aspa-orange-5: #d78840;--mantine-color-aspa-orange-6: #d68032;--mantine-color-aspa-orange-7: #bd6d24;--mantine-color-aspa-orange-8: #a9611d;--mantine-color-aspa-orange-9: #945212;--mantine-color-aspa-brown-0: #faf5ef;--mantine-color-aspa-brown-1: #f0e8e0;--mantine-color-aspa-brown-2: #e2cdbb;--mantine-color-aspa-brown-3: #d5b192;--mantine-color-aspa-brown-4: #c9996f;--mantine-color-aspa-brown-5: #c38a59;--mantine-color-aspa-brown-6: #c1834d;--mantine-color-aspa-brown-7: #a96f3d;--mantine-color-aspa-brown-8: #986335;--mantine-color-aspa-brown-9: #845429;--mantine-color-aspa-gray-0: #f4f5f5;--mantine-color-aspa-gray-1: #e7e7e7;--mantine-color-aspa-gray-2: #cdcdcd;--mantine-color-aspa-gray-3: #b1b1b1;--mantine-color-aspa-gray-4: #999999;--mantine-color-aspa-gray-5: #8a8a8a;--mantine-color-aspa-gray-6: #838383;--mantine-color-aspa-gray-7: #707071;--mantine-color-aspa-gray-8: #636466;--mantine-color-aspa-gray-9: #52575b;}:root[data-mantine-color-scheme="dark"]{--mantine-color-aspa-orange-text: var(--mantine-color-aspa-orange-4);--mantine-color-aspa-orange-filled: var(--mantine-color-aspa-orange-8);--mantine-color-aspa-orange-filled-hover: var(--mantine-color-aspa-orange-9);--mantine-color-aspa-orange-light: rgba(214, 128, 50, 0.15);--mantine-color-aspa-orange-light-hover: rgba(214, 128, 50, 0.2);--mantine-color-aspa-orange-light-color: var(--mantine-color-aspa-orange-3);--mantine-color-aspa-orange-outline: var(--mantine-color-aspa-orange-4);--mantine-color-aspa-orange-outline-hover: rgba(220, 151, 88, 0.05);--mantine-color-aspa-brown-text: var(--mantine-color-aspa-brown-4);--mantine-color-aspa-brown-filled: var(--mantine-color-aspa-brown-8);--mantine-color-aspa-brown-filled-hover: var(--mantine-color-aspa-brown-9);--mantine-color-aspa-brown-light: rgba(193, 131, 77, 0.15);--mantine-color-aspa-brown-light-hover: rgba(193, 131, 77, 0.2);--mantine-color-aspa-brown-light-color: var(--mantine-color-aspa-brown-3);--mantine-color-aspa-brown-outline: var(--mantine-color-aspa-brown-4);--mantine-color-aspa-brown-outline-hover: rgba(201, 153, 111, 0.05);--mantine-color-aspa-gray-text: var(--mantine-color-aspa-gray-4);--mantine-color-aspa-gray-filled: var(--mantine-color-aspa-gray-8);--mantine-color-aspa-gray-filled-hover: var(--mantine-color-aspa-gray-9);--mantine-color-aspa-gray-light: rgba(131, 131, 131, 0.15);--mantine-color-aspa-gray-light-hover: rgba(131, 131, 131, 0.2);--mantine-color-aspa-gray-light-color: var(--mantine-color-aspa-gray-3);--mantine-color-aspa-gray-outline: var(--mantine-color-aspa-gray-4);--mantine-color-aspa-gray-outline-hover: rgba(153, 153, 153, 0.05);}:root[data-mantine-color-scheme="light"]{--mantine-color-aspa-orange-text: var(--mantine-color-aspa-orange-filled);--mantine-color-aspa-orange-filled: var(--mantine-color-aspa-orange-6);--mantine-color-aspa-orange-filled-hover: var(--mantine-color-aspa-orange-7);--mantine-color-aspa-orange-light: rgba(214, 128, 50, 0.1);--mantine-color-aspa-orange-light-hover: rgba(214, 128, 50, 0.12);--mantine-color-aspa-orange-light-color: var(--mantine-color-aspa-orange-6);--mantine-color-aspa-orange-outline: var(--mantine-color-aspa-orange-6);--mantine-color-aspa-orange-outline-hover: rgba(214, 128, 50, 0.05);--mantine-color-aspa-brown-text: var(--mantine-color-aspa-brown-filled);--mantine-color-aspa-brown-filled: var(--mantine-color-aspa-brown-6);--mantine-color-aspa-brown-filled-hover: var(--mantine-color-aspa-brown-7);--mantine-color-aspa-brown-light: rgba(193, 131, 77, 0.1);--mantine-color-aspa-brown-light-hover: rgba(193, 131, 77, 0.12);--mantine-color-aspa-brown-light-color: var(--mantine-color-aspa-brown-6);--mantine-color-aspa-brown-outline: var(--mantine-color-aspa-brown-6);--mantine-color-aspa-brown-outline-hover: rgba(193, 131, 77, 0.05);--mantine-color-aspa-gray-text: var(--mantine-color-aspa-gray-filled);--mantine-color-aspa-gray-filled: var(--mantine-color-aspa-gray-6);--mantine-color-aspa-gray-filled-hover: var(--mantine-color-aspa-gray-7);--mantine-color-aspa-gray-light: rgba(131, 131, 131, 0.1);--mantine-color-aspa-gray-light-hover: rgba(131, 131, 131, 0.12);--mantine-color-aspa-gray-light-color: var(--mantine-color-aspa-gray-6);--mantine-color-aspa-gray-outline: var(--mantine-color-aspa-gray-6);--mantine-color-aspa-gray-outline-hover: rgba(131, 131, 131, 0.05);}
  </style>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="m_7485cace mantine-Container-root"
    data-fluid="true"
  >
    <style
      data-mantine-styles="inline"
    >
      .__m__-«rr»{--grid-gutter:var(--mantine-spacing-xl);}
    </style>
    <div
      class="m_410352e9 mantine-Grid-root __m__-«rr»"
    >
      <div
        class="m_dee7bd2f mantine-Grid-inner"
      >
        <style
          data-mantine-styles="inline"
        >
          .__m__-«rt»{--col-flex-grow:auto;--col-flex-basis:100%;--col-max-width:100%;}@media(min-width: 48em){.__m__-«rt»{--col-flex-grow:auto;--col-flex-basis:50%;--col-max-width:50%;}}
        </style>
        <div
          class="m_96bdd299 mantine-Grid-col __m__-«rt»"
        >
          <h1
            class="title m_8a5d1357 mantine-Title-root"
            data-order="1"
            style="--title-fw: var(--mantine-h1-font-weight); --title-lh: var(--mantine-h1-line-height); --title-fz: var(--mantine-h1-font-size);"
          >
            ASPA - Auckland Student Pool Association
          </h1>
          <p
            class="mantine-focus-auto text m_b6d8b162 mantine-Text-root"
          >
            Leque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, Leque porro quisquam est qui dolorem ipsum quia dolor sit amet
          </p>
          <div
            class="m_4081bf90 mantine-Group-root"
            style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: flex-start; --group-wrap: wrap;"
          >
            <button
              class="mantine-focus-auto mantine-active button m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
              style="--button-color: var(--mantine-color-white);"
              type="button"
            >
              <span
                class="m_80f1301b mantine-Button-inner"
              >
                <span
                  class="m_811560b9 mantine-Button-label"
                >
                  View ASPA Page
                </span>
              </span>
            </button>
            <p
              class="mantine-focus-auto group m_b6d8b162 mantine-Text-root"
            >
              View ASPA Page
            </p>
          </div>
        </div>
        <style
          data-mantine-styles="inline"
        >
          .__m__-«r15»{--col-flex-grow:auto;--col-flex-basis:100%;--col-max-width:100%;}@media(min-width: 48em){.__m__-«r15»{--col-flex-grow:auto;--col-flex-basis:50%;--col-max-width:50%;}}
        </style>
        <div
          class="m_96bdd299 mantine-Grid-col imageContainer __m__-«r15»"
        >
          <img
            alt="ASPA Title"
            class="image hoverUpDown"
            data-nimg="1"
            decoding="async"
            height="0"
            loading="lazy"
            src="/aspa_title.svg"
            style="color: transparent;"
            width="0"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
